gomock/call.go

```diff
@@ -192,27 +193,30 @@ func (c *Call) String() string {
 }
 
 // Tests if the given call matches the expected call.
-func (c *Call) matches(args []interface{}) bool {
+// If yes, returns nil. If no, returns error with message explaining why it does not match.
+func (c *Call) matches(args []interface{}) error {
 	if len(args) != len(c.args) {
-		return false
+		return fmt.Errorf("Invalid number of arguments of call: %s. Set: %s, while this call takes: %s",
```

Comment:
All the error messages returned are worded as error messages to their respective functions. This makes sense in context but I think it doesn't result in the best final error message to the user. Perhaps, instead, they could be worded for the final use. For example. The errors could say something like "Expected call &lt;call details&gt; &lt;mismatch&gt;". Then, the final message will read something like:
Unexpected call &lt;call details&gt; because:
Expected call &lt;call details&gt; has the wrong number of arguments (&lt;got&gt; vs. &lt;want&gt;)
Expected call &lt;call details&gt; doesn't match argument #&lt;index&gt; (&lt;got&gt; vs &lt;want&gt;)

Let me know what you think.